plugins/plugin.go

```diff
@@ -141,9 +141,12 @@ func (p *Plugin) StartPlugin() error {
 // Metrics polls the DC/OS components and returns a slice of
 // producers.MetricsMessage.
 func (p *Plugin) Metrics() ([]producers.MetricsMessage, error) {
-	p.Log.Info("Getting metrics from metrics service")
 	metricsMessages := []producers.MetricsMessage{}
+	if len(p.AuthToken) == 0 {
+		return metricsMessages, errors.New("Auth token must be set, use --auth-token <token>")
```

Comment:
Probably can just return nil for the first return value here.  Also, maybe move this check above the line

`metricsMessages := []producers.MetricsMessage{}`